patches/api/0352-Patch-Add-new-overload-to-PersistentDataContainer-ha.patch

```diff
@@ -0,0 +1,30 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: u9g <winworkswow@gmail.com>
+Date: Mon, 3 Jan 2022 23:27:21 -0500
+Subject: [PATCH] [Patch] Add new overload to PersistentDataContainer#has
+
+Adds the new overload: PersistentDataContainer#has(NamespacedKey key)
+
+diff --git a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+index bf2a957be52d86d07d7d303c86bd3da5fe0a16c0..071cf222508ee15d95d7aa76f7979d83c4a6e421 100644
+--- a/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
++++ b/src/main/java/org/bukkit/persistence/PersistentDataContainer.java
+@@ -151,4 +151,18 @@ public interface PersistentDataContainer {
+      */
+     @NotNull
+     PersistentDataAdapterContext getAdapterContext();
++
++    // Paper start
++    /**
++     * Returns if the persistent metadata provider has metadata registered
++     * matching the provided key.
++     * 
++     * @param key the key to look up in the custom tag map
```

Comment:
The concept of a custom tag map is pretty unknown to the API.
This should probably be rephrased to something along the lines of
````
@param key the key which existence should be checked inside the persistent data container.
```

Reply:
Done :D